music_assistant/server/providers/jellyfin/__init__.py

```diff
@@ -419,31 +419,37 @@ async def _parse_track(self, jellyfin_track: JellyMediaItem) -> Track:
                         artist_item[ITEM_KEY_NAME],
                     )
                 )
-        elif ITEM_KEY_ALBUM_ID in jellyfin_track:
-            parent_album = await self._client.get_item(jellyfin_track[ITEM_KEY_ALBUM_ID])
-            if ITEM_KEY_ALBUM_ARTISTS in parent_album:
-                for artist_item in parent_album[ITEM_KEY_ALBUM_ARTISTS]:
-                    track.artists.append(
-                        self._get_item_mapping(
-                            MediaType.ARTIST,
-                            artist_item[ITEM_KEY_ID],
-                            artist_item[ITEM_KEY_NAME],
+
+        if ITEM_KEY_ALBUM_ID in jellyfin_track:
+            if ITEM_KEY_ALBUM in jellyfin_track:
+                track.album = self._get_item_mapping(
+                    MediaType.ALBUM,
+                    jellyfin_track[ITEM_KEY_ALBUM_ID],
+                    jellyfin_track[ITEM_KEY_ALBUM],
+                )
+
+            # If we have an AlbumID not have not managed to set track.artists or track.album
+            # Let's see if we can pull it from the album
```

Comment:
Isn't this a mega bug in Jellyfin (or its api) ?
Track artists are really something different than album artists.

What will happen with a compilation album for example ?
Are the tracks going to have "Various artists" as artist ?

Reply:
No, not a jellyfin bug. Jellyfin has separate metadata for both.

Existing code in the provider is trying to handle the case where we have an album artist and not a track artist by assuming the album artist is more useful than "Various".

In my library it would be more right than wrong, but not a fan personally of making fallback guesses. I'd rather fail it or set it to unknown.

This is preserving current behavior. Planning on sorting out unknown/various artists in a follow up PR, if that's ok?

Reply:
OK, ended up adding the [unknown] artist we talked about on Discord.